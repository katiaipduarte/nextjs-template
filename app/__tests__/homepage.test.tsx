import '@testing-library/jest-dom'

import { render, screen } from '@testing-library/react'

import Home from '../page'

describe('Home', () => {
	it('renders homepage unchanged', () => {
		const { container } = render(<Home />)
		expect(container).toMatchSnapshot()
	})

	it('renders a heading', () => {
		render(<Home />)

		const heading = screen.getByRole('heading', {
			name: /my normal next\.js page with static metadata/iu,
		})

		expect(heading).toBeInTheDocument()
	})
})
