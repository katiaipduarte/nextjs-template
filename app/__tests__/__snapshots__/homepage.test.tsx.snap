// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-between p-24"
  >
    <h1>
      My Normal Next.js Page with Static Metadata
    </h1>
    <a
      href="/about"
    >
      <h2>
        About Page â†’
      </h2>
      <p>
        Playwright will test if this link is working.
      </p>
    </a>
    <a
      href="/hydration-stream-suspense"
    >
      (recommended method): React Query with Stream Hydration --- Bad for SEO with good performance
    </a>
  </main>
</div>
`;
